From: Chris Morgan <macroalpha82@gmail.com>
Date: Mon, 30 Mar 2026 14:58:53 -0500
Subject: arm64: dts: rockchip: Add icm42607p IMU for RG-DS

From: Chris Morgan <macromorgan@hotmail.com>

Add the Invensense ICM42607P IMU for the Anbernic RG-DS. Mount-matrix
was tested with iio-sensor-proxy and reports correct orientation.

Updated for kernel 7.0 DTS where the accel pinctrl is already upstream
and the aw87391/icm42607p entries are comment stubs.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
[aenertia@aenertia.net: rebased for kernel 7.0 DTS changes]
---
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-anbernic-rg-ds.dts b/arch/arm64/boot/dts/rockchip/rk3568-anbernic-rg-ds.dts
--- a/arch/arm64/boot/dts/rockchip/rk3568-anbernic-rg-ds.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3568-anbernic-rg-ds.dts
@@ -843,7 +843,18 @@
 	/* awinic,aw87391 at 0x58 */
 	/* awinic,aw87391 at 0x5b */
-	/* invensense,icm42607p at 0x68 */
+	icm42607p: imu@68 {
+		compatible = "invensense,icm42607p";
+		reg = <0x68>;
+		interrupt-names = "INT1";
+		interrupt-parent = <&gpio0>;
+		interrupts = <RK_PD6 IRQ_TYPE_EDGE_FALLING>;
+		mount-matrix = "-1", "0", "0",
+			       "0", "1", "0",
+			       "0", "0", "-1";
+		pinctrl-0 = <&accel_irq>;
+		pinctrl-names = "default";
+	};
 };
 
 &i2c3 {
